src/HexagonOptimize.cpp

```diff
@@ -1079,11 +1081,59 @@ class OptimizePatterns : public IRMutator {
             // that they generate.
             internal_assert(op->args.size() == 3);
             return mutate(lower_lerp(op->args[0], op->args[1], op->args[2]));
+        } else if (op->is_intrinsic(Call::div_round_to_zero) &&
+                   !op->type.is_float() && op->type.lanes() > 1) {
+            internal_assert(op->args.size() == 2);
+            Expr a = mutate(op->args[0]);
+            Expr b = mutate(op->args[1]);
+            debug(1) << "Using long division: (num: " << a << "); (den: " << b
+                     << ")\n";
+            if (a.type().is_uint()) {
```

Comment:
Does `Call::div_round_to_zero` require that a and b have identical types? (If not, this call to `unsigned_long_div` could assert-fail)

Reply:
Right now all the call::div_round_to_zero are generated through lower_euclidean_div which also needs a and b to have same types. All calls to div_round_to_zero should have the same types here.

Reply:
In that case, we should probably assert in the other code path? It's probably not getting tested?

Also, if you do my other suggestions re: adding a helper for signed division, maybe that should clarify e.g. `long_div_round_to_zero` or something.